docs/hoon/twig/cen-call/sig-open.md

```diff
@@ -13,6 +13,10 @@ sort: 3
 :rap(p :make(q +6 r))
 ```
 
+```
+=<(p %=(q +6 r))
+```
+
```

Comment:
Yeah.  Basically, Joe, you're taking this exercise quite a bit more
seriously than I did.  That's good, of course, but...

The original idea was that the macro expansions in ++open would be
readable.  Unfortunately they're... somewhat readable.  They do work
perfectly.  Of course.

But these expansions in the doc were always pseudocode.  As pseudocode, I
was a bit sloppy and loose with them.  There used to be comments in the
hoon.hoon source that tried to establish a rigorous pseudocode for defining
Hoon macros, but there's no such thing as a rigorous pseudocode, eh?

Also, the undocumented runes you've discovered were undocumented on purpose
-- they're (generally) deprecated.  That's fine.  Not documenting something
is a pretty cheesy way of deprecating it.

You're obviously contributing on a pretty high level here, so I encourage
you to try and think hard about how exactly this should be done!  Just
filling you in on a bit of the history...

On Thu, Jun 23, 2016 at 11:32 AM, Joe Bryan <notifications@github.com>
wrote:

> In docs/hoon/twig/cen-call/sig-open.md
> <https://github.com/urbit/docs/pull/55#discussion_r68289942>:
>
> > @@ -13,6 +13,10 @@ sort: 3
> >  :rap(p :make(q +6 r))
> >  ```
> >
> > +```
> > +=<(p %=(q +6 r))
> > +```
> > +
>
> yeah, definitely.
>
> My point is just that it wasn't clear to me, reading that expansion, that
> q needs to be a "reference" to an arm defined elsewhere, and it took a
> lot of trial and error to figure that out.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/urbit/docs/pull/55/files/fb89b530694cb031b6e33d0b360f304e6cb77089#r68289942>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALyARTdTYx_Uvuyy6Tof1bsR6aW1_VJks5qOtE1gaJpZM4I9D6c>
> .
>